xbmc/pvr/channels/PVRChannel.cpp

```diff
@@ -24,6 +24,10 @@
 #include "utils/Variant.h"
 #include "utils/log.h"
 
+#include "PVRChannelGroupsContainer.h"
+#include "PVRChannelGroups.h"
+#include "PVRChannelGroup.h"
```

Comment:
This is completely wrong as it breaks the object hierarchy. Channel must not depend on any of the channel containers.